Pass the correct length of file bodies to clang

mheinzler opened the pull request:
We need to pass the length of the encoded bytes array to clang, not the
length of the string which would be shorter if it includes symbols
outside of ASCII.

Fixes #443.

mheinzler commented:
It seems that the build errors are not related to these changes.

niosus commented:
Makes sense. The errors look super weird. I will try to fix them on the weekend and will merge your PR then.
Consider contributing it to llvm folks too.

mheinzler commented:
Do you think this is an issue in llvm/clang?
I think the problem simply is that clang received the wrong length for the bytes array. Because of that it sees the file as cut off which results in all kinds of errors about missing brackets and so on.

niosus commented:
Well, it's definitely an issue with their Python bindings and the ones packaged here are just a slightly modified copy of those.

mheinzler commented:
Ah, now I understand. I didn't know that you stole them from somewhere :smile:.

While looking at the original llvm code I noticed that this encoding issue affects all uses of file contents. For example code completion also doesn't work correctly when the file contains multi-byte characters.

What do you think of this new commit which addresses all these issues? I also removed the UTF-8 encoding in reparse in favor of the bytes representation to match the behaviour of the other sections.

If you agree I can squash the commits for you to merge and will also forward this to the llvm people.

niosus commented:
I can also squash the commits while merging. The only thing I would ask you is to rebase on current dev again as I have fixed the issues with the build system and I am more comfortable merging green PRs :)

mheinzler commented:
Here you go. I will also send this to the llvm team.
